### ISSUE TITLE
NLU Evaluation - error when writing results back to spreadsheet

### ISSUE THREAD

**Issue by AngryBalloon** [GoogleCloudPlatform/contact-center-ai-samples#541]
I've been trying to use nlu-evaluation/nlu_evaluation.ipynb to evaluate if there will be any changes when our bots are changed from standard to advanced NLU type.

When running the "Fill and Run" part it connects to the spreadsheet to read data, run evaluations successfully, but when writing results back to the spreedsheet the error below is thrown. At the same time spreadsheet is wiped for all data, the only thing that remains are the two sheets Agent Dataset and Summary.

Console output:
```
Using API endpoint `europe-west3-dialogflow.googleapis.com:443`
Worksheet `Summary` does not exist. Creating...
Getting flow mapping...done
Found 1 flows.
Getting page mapping...done
Started evaluation.
Evaluated 2 / 2 utterances.
---------------------------------------------------------------------------
APIError                                  Traceback (most recent call last)
[<ipython-input-5-0c65015985a5>](https://localhost:8080/#) in <cell line: 33>()
     31 creds, _ = default()
     32 print(f"### Running evaluation on agent {agent_label_1} ###\n")
---> 33 run(creds, agent_label_1, project_id_1, region_1, agent_id_1, language_code_1, spreadsheet_url, dataset_tab, summary_tab)
     34 
     35 if use_second_agent:

4 frames
[<ipython-input-4-976eabb0be55>](https://localhost:8080/#) in run(creds, agent_label, project_id, region, agent_id, language_code, spreadsheet_url, dataset_tab, summary_tab)
    275   dataset = evaluate_dataset(session_client, flow_client, page_client, dataset, agent_label, project_id, region, agent_id, language_code)
    276 
--> 277   write_dataset(spreadsheet_info['dataset_worksheet'], dataset)
    278   write_summary(spreadsheet_info['summary_worksheet'], dataset, agent_label, timestamp, dataset_tab)

[<ipython-input-4-976eabb0be55>](https://localhost:8080/#) in write_dataset(dataset_worksheet, dataset)
    251 
    252   dataset_worksheet.clear()
--> 253   dataset_worksheet.update(range_name=([dataset.columns.values.tolist()] + dataset.values.tolist()), values=None)
    254 
    255 

[/usr/local/lib/python3.10/dist-packages/gspread/worksheet.py](https://localhost:8080/#) in update(self, values, range_name, raw, major_dimension, value_input_option, include_values_in_response, response_value_render_option, response_date_time_render_option)
   1237         }
   1238 
-> 1239         response = self.client.values_update(
   1240             self.spreadsheet_id,
   1241             full_range_name,

[/usr/local/lib/python3.10/dist-packages/gspread/http_client.py](https://localhost:8080/#) in values_update(self, id, range, params, body)
    166         """
    167         url = SPREADSHEET_VALUES_URL % (id, quote(range))
--> 168         r = self.request("put", url, params=params, json=body)
    169         return r.json()
    170 

[/usr/local/lib/python3.10/dist-packages/gspread/http_client.py](https://localhost:8080/#) in request(self, method, endpoint, params, data, json, files, headers)
    121             return response
    122         else:
--> 123             raise APIError(response)
    124 
    125     def batch_update(self, id: str, body: Optional[Mapping[str, Any]]) -> Any:

APIError: {'code': 400, 'message': "Unable to parse range: 'Agent Dataset'![['Flow Display Name', 'Page Display Name', 'Utterance', 'Expected Intent', 'Frequency', 'Detected Intent TestBot1', 'Confidence TestBot1', 'Results TestBot1'], ['Default Start Flow', 'Start Page', 'bil', 'test.intent.1', 1, 'test.intent.1', 1.0, 'PASS'], ['Default Start Flow', 'Start Page', 'mc', 'test.intent.2', 1, 'test.intent.2', 1.0, 'PASS']]", 'status': 'INVALID_ARGUMENT'}
```

Spreadsheet before run:
![image](https://github.com/GoogleCloudPlatform/contact-center-ai-samples/assets/37542885/d09dd1c3-fbc0-45d7-87af-923ed2d285d9)

**Comment by koverholt** [comment]
@AngryBalloon, thanks for letting us know. I fixed the issue in PR #543 and tested, the notebook should behave as expected now and write the output to the appropriate `Agent Dataset` and `Summary` sheets.

**Comment by AngryBalloon** [comment]
Thanks @koverholt, I can confirm that it works from my end as well!

And thanks for sharing that link :)
